subprojects/model-core/src/main/java/org/gradle/internal/reflect/validation/TypeValidationProblemRenderer.java

```diff
@@ -76,7 +76,7 @@ public static String renderMinimalInformationAbout(TypeValidationProblem problem
      * a better way to display multiline deprecation warnings
      */
     public static String convertToSingleLine(String message) {
-        return message.replaceAll("(\\r?\\n)+", ". ")
+        return message.replaceAll("(\\r?\\n *)+", ". ")
```

Comment:
Why is this change needed? What about `\t`? Can we use `\s*` instead?

Reply:
For the assertions, we convert the multiline error to the single line deprecation message. This change allows a multiline _indented_ error to be converted correctly to the single line deprecation message. This allowed me to get rid of the method which removed the indentation.

I wouldn't add any more regex logic here as long as the joining work and remove this method as soon as we have multiline deprecation messages.